client/src/elm/Pages/MessagingCenter/Utils.elm

```diff
@@ -108,11 +108,48 @@ surveyQuestionsAnswered surveyType surveyForm =
     Dict.size surveyForm == List.length surveyQuestions
 
 
-resolveSurveyScoreDialogState : ResilienceSurveyType -> Int -> SurveyScoreDialogState
-resolveSurveyScoreDialogState surveyType score =
+resolveSurveyScoreDialogState : NurseId -> ResilienceSurveyType -> Int -> ModelIndexedDb -> SurveyScoreDialogState
+resolveSurveyScoreDialogState nurseId surveyType score db =
     case surveyType of
         ResilienceSurveyQuarterly ->
             QuarterlySurveyScore score
 
         ResilienceSurveyAdoption ->
-            AdoptionSurveyScore score
+            let
+                surveys =
+                    Dict.get nurseId db.resilienceSurveysByNurse
+                        |> Maybe.andThen RemoteData.toMaybe
+                        |> Maybe.map Dict.values
+                        |> Maybe.withDefault []
+
+                adoptionSurveys =
+                    List.filter (\survey -> survey.surveyType == ResilienceSurveyAdoption) surveys
+
+                sortedSurveys =
+                    List.sortWith (\survey1 survey2 -> Date.compare survey1.dateMeasured survey2.dateMeasured) adoptionSurveys
+
+                toScore answer =
+                    case answer of
+                        ResilienceSurveyQuestionOption0 ->
+                            1
+
+                        ResilienceSurveyQuestionOption1 ->
+                            2
+
+                        ResilienceSurveyQuestionOption2 ->
+                            3
+
+                        ResilienceSurveyQuestionOption3 ->
+                            4
+
+                        ResilienceSurveyQuestionOption4 ->
+                            5
+
+                previousSurveysScores =
```

Comment:
In our system, sometimes we see that measurements are saved more than once.
For example, when taking height measurement, sometimes we get 2 duplicate nodes instead of one.
This may also happen when we save a survey.
Therefore, lets make sure we take correct surveys - make sure surveys got a unique `dateMeasured`.
Use `List.foldl`.

So, correct order of actions would be:
1. Get surveys with unique `dateMeasured`.
2. Take 2.
3. Compute score.